ozaria/site/components/teacher-dashboard/BaseCurriculumGuide/components/ModalCapstoneProject.vue

```diff
@@ -0,0 +1,172 @@
+<script>
+import { mapGetters, mapMutations } from 'vuex'
+import Modal from 'ozaria/site/components/common/BaseModalContainer.vue'
+
+export default Vue.extend({
+  components: {
+    Modal,
+  },
+  data () {
+    return {
+      goal: {
+        'introduction-to-computer-science': 'Write a program that draws a dragon.',
+        'computer-science-2': 'Create a text-based adventure game.',
+        'computer-science-3': 'Create a simulation.',
+        'computer-science-4': 'Recreate an arcade game, card game, or board game.',
+      },
+      requirements: {
+        'introduction-to-computer-science': ['Your dragon must be at least half the width and height of your viewing window.', 'You must meaningfully use two variables in your program.'],
+        'computer-science-2': ['Your game includes at least 3 questions that a user can answer.', 'Different output is generated depending on the input provided by the user.', 'You meaningfully define and call a function that uses at least one parameter.'],
+        'computer-science-3': ['At least one input must be randomly generated.', 'Output must change depending on the input randomly generated.', 'Define and call a function that uses at least one parameter and one return statement.'],
+        'computer-science-4': ['Users should be able to provide input in order to play the game, and this input should influence game play in some way.', 'Use at least one list in your program.', 'Define and call a function that uses at least one parameter and one return statement.'],
+      },
+    }
+  },
+  computed: {
+    ...mapGetters({
+      visible: 'baseCurriculumGuide/isCapstoneModalVisible',
```

Comment:
we want to re-use this modal outside of curriculum guide in student side too. Can we instead rely on props being passed for the correct content to show?